zebra-consensus/src/transaction.rs

```diff
@@ -554,17 +565,20 @@ where
             // Get the `value_balance` to calculate the transaction fee.
             let value_balance = tx.value_balance(&spent_utxos);
 
+            let zip233_amount = match *tx {
+                Transaction::V6{ .. } => tx.zip233_amount(),
+                _ => Amount::zero()
+            };
+
             // Calculate the fee only for non-coinbase transactions.
             let mut miner_fee = None;
             if !tx.is_coinbase() {
                 // TODO: deduplicate this code with remaining_transaction_value()?
-                miner_fee = match value_balance {
-                    Ok(vb) => match vb.remaining_transaction_value() {
-                        Ok(tx_rtv) => Some(tx_rtv),
-                        Err(_) => return Err(TransactionError::IncorrectFee),
-                    },
-                    Err(_) => return Err(TransactionError::IncorrectFee),
-                };
+                miner_fee = value_balance
+                    .map(|vb| vb.remaining_transaction_value())
+                    .map(|tx_rtv| tx_rtv - zip233_amount)
```

Comment:
One thing I'm not seeing that I could have missed is a test that exercises the logic here - do we have this anywhere?

Reply:
There is a `nu7_nsm_transactions()` test which should call the function above.

Reply:
Ah, great. Thanks.